crates/muvm/src/guest/net.rs

```diff
@@ -33,63 +116,135 @@ pub fn configure_network() -> Result<()> {
         sethostname(hostname.as_bytes()).context("Failed to set hostname")?;
     }
 
-    let dhcpcd_path = find_in_path("dhcpcd").context("Failed to check existence of `dhcpcd`")?;
-    let dhcpcd_path = if let Some(dhcpcd_path) = dhcpcd_path {
-        Some(dhcpcd_path)
-    } else {
-        find_executable("/sbin/dhcpcd").context("Failed to check existence of `/sbin/dhcpcd`")?
-    };
-    if let Some(dhcpcd_path) = dhcpcd_path {
-        let output = Command::new(dhcpcd_path)
-            .args(["-M", "--nodev", "eth0"])
-            .output()
-            .context("Failed to execute `dhcpcd` as child process")?;
-        debug!(output:?; "dhcpcd output");
-        if !output.status.success() {
-            let err = if let Some(code) = output.status.code() {
-                anyhow!("`dhcpcd` process exited with status code: {code}")
-            } else {
-                anyhow!(
-                    "`dhcpcd` process terminated by signal: {}",
-                    output
-                        .status
-                        .signal()
-                        .expect("either one of status code or signal should be set")
-                )
-            };
-            Err(err)?;
-        }
+    let (rtnl, _) = NlRouter::connect(NlFamily::Route, None, Groups::empty())?;
```

Comment:
Should I create some intermediate functions? I thought I already abstracted enough by separating those netlink operations. I could probably move the DHCP stuff to its own function and the post-SLAAC stuff as well, but it's not really practical to have a separate "SLAAC" function because some steps need to be done before sending DHCP requests.

To me it's actually easier to follow if it's sequential/flattened, because adding and removing routes and addresses is intertwined with some other operations, but I guess I see the case for intermediate functions (even though I don't find anything wrong with a configure_network() function setting interface flags).

Reply:
All fair points. I also favour not making unnecessary abstractions.

Reply:
I actually tried splitting stuff in two further functions, it looks okay I guess... I'll post that too once I figure out why muvm just stopped working entirely for me (even without my changes).